akka-http/src/main/scala/akka/http/scaladsl/server/HttpApp.scala

```diff
@@ -134,7 +134,7 @@ abstract class HttpApp extends Directives {
   protected def waitForShutdownSignal(system: ActorSystem)(implicit ec: ExecutionContext): Future[Done] = {
     val promise = Promise[Done]()
     sys.addShutdownHook {
-      promise.success(Done)
+      promise.trySuccess(Done)
```

Comment:
Line 141 could be also replaced with `trySuccess`.
I guess you are right about the `blocking { }` part, the whole purpose of this line is to block.

I'll apply the feedback